### DISCUSSION TITLE
Override PrimaryButtonStyle in ContentDialog

### DISCUSSION THREAD

**Discussion by HakanL** [unoplatform/uno#12833] [Q&A]
I realize I'm a novice when it comes to XAML, but how can I override the style of the buttons on a ContentDialog? I've tried several different ways, but the style is the same. I just want my buttons (Primary and Secondary) to be taller (so they are easier to click on when using a touch screen). Platform is Skia, but I assume that shouldn't matter. I know I can set the PrimaryButtonStyle, but nothing happens, I've tried to change the background, border, height, nothing seems to "take".

**Comment by Xiaoy312** [comment]
Sorry for the late reply.
We are not able to reproduce the issue you are facing. It would be great if you can provide a repro sample, or shortly describe your xaml/cs setup to help us locate the issue?
Here is a simple a setup that is working on skia:
```xml
<Page.Resources>
	<Style x:Key="BasicButtonStyle" TargetType="Button">
		<Setter Property="Background" Value="Red" />
		<Setter Property="Padding" Value="4,16" />
		<Setter Property="Template">
			<Setter.Value>
				<ControlTemplate TargetType="Button">
					<Border Background="{TemplateBinding Background}"
							Padding="{TemplateBinding Padding}">
						<ContentPresenter Content="{TemplateBinding Content}" />
					</Border>
				</ControlTemplate>
			</Setter.Value>
		</Setter>
	</Style>
</Page.Resources>
<Grid>
	<Button Content="ShowDialog" Click="ShowDialog" />
</Grid>
```
```cs
	private async void ShowDialog(object sender, RoutedEventArgs e)
	{
		var dialog = new ContentDialog()
		{
			Title = "TItle",
			Content = "blablabla",
			PrimaryButtonText = "Yes",
			CloseButtonText = "Dismiss",
			PrimaryButtonStyle = (Style)this.Resources["BasicButtonStyle"],
		};

		await dialog.ShowAsync();
	}
```
result:
![image](https://github.com/unoplatform/uno/assets/2359550/5af60a8c-e51d-4c45-8e0a-cad87b9d0d5f)

**Reply by HakanL** [reply]
Oddly enough it seems to work unless I set `DefaultButton="Primary"` in the ContentDialog. Would that change the style somehow?
